### PR TITLE
ZFIN-8344: Refactor Mutant Repository for Hibernate

### PR DESCRIPTION
**Pull request by rtaylorzfin**

### CODE REVIEW

File: source/org/zfin/mutant/repository/HibernateMutantRepository.java [hunk 1]
```diff
@@ -170,203 +165,206 @@ public Object transformTuple(Object[] tuple, String[] aliases) {
...
+                        WHERE genofeat.feature = :feature 
+                        AND genofeat.genotype = geno 
+                        ORDER BY geno.nameOrder 
+                        """;
 
-        String hql = "select  distinct geno from Genotype geno, GenotypeFeature genofeat " +
-            "WHERE  genofeat.feature = :feature " +
-            "AND genofeat.genotype =geno " +
-            "ORDER BY geno.nameOrder";
+                    Query query = session.createQuery(hql);
```
Comment (cmpich) [inline source/org/zfin/mutant/repository/HibernateMutantRepository.java:234]: Retrieve generic Query
query<Genotype> query = session.createQuery(hql, Genotype.class)

Comment (rtaylorzfin) [reply source/org/zfin/mutant/repository/HibernateMutantRepository.java]: Thanks for the comment.  I added types to each of those queries.

File: source/org/zfin/mutant/repository/HibernateMutantRepository.java [hunk 2]
```diff
@@ -276,97 +271,102 @@ public PaginationResult<FishGenotypeFeature> getFishByFeature(String featureId,
...
+            image member of fig.images AND 
+            res.expressionFound = true AND 
+            exp.fishExperiment.fish.genotype.zdbID = :genoZdbID 
+            """;
+        Query query = session.createQuery(hql, Number.class);
         query.setParameter("term", term);
-        query.setString("genoZdbID", genotype.getZdbID());
+        query.setParameter("genoZdbID", genotype.getZdbID());
 
         return ((Number) query.uniqueResult()).intValue();
```
Comment (cmpich) [inline source/org/zfin/mutant/repository/HibernateMutantRepository.java:351]: The cast is unneccessary...

File: source/org/zfin/mutant/repository/HibernateMutantRepository.java [hunk 3]
```diff
@@ -276,97 +271,102 @@ public PaginationResult<FishGenotypeFeature> getFishByFeature(String featureId,
...
+                                              ExpressionExperiment exp 
+            where 
+            res member of exp.expressionResults AND 
+            res.entity.superterm = :term AND 
+            fig member of res.figures AND 
+            image member of fig.images AND 
+            res.expressionFound = true AND 
+            exp.fishExperiment.fish.genotype.zdbID = :genoZdbID 
+            """;
+        Query query = session.createQuery(hql, Number.class);
```
Comment (cmpich) [inline source/org/zfin/mutant/repository/HibernateMutantRepository.java:347]: use the generic Query class, i.e.
Query<Number> query =session.createQuery(hql, Number.class);

**Review by cmpich — COMMENTED** [review]
Make sure to use the generic Query class then you do not need to cast the return of the query.

File: source/org/zfin/mutant/repository/HibernateMutantRepository.java (hunk 3 above)
Comment (rtaylorzfin) [reply source/org/zfin/mutant/repository/HibernateMutantRepository.java:347]: Sorry, I'm not seeing the distinction?

**Review by cmpich — APPROVED** [review]
Looks good now.

**Pull request merged**